src/bmi_cfe.c

```diff
@@ -1787,6 +1786,10 @@ static int Get_var_nbytes (Bmi *self, const char *name, int * nbytes)
     for (i = 0; i < INPUT_VAR_NAME_COUNT; i++) {
         if (strcmp(name, input_var_names[i]) == 0) {
             item_count = input_var_item_count[i];
+	    if (strcmp(name, "soil_moisture_profile") == 0 || strcmp(name, "soil_layer_depths_m") == 0) {
+	      cfe_state_struct *cfe_ptr;
+	      item_count = ((cfe_state_struct *)(self->data))->soil_reservoir.n_soil_layers + 1;
```

Comment:
why is this +1?

Reply:
I think this +1 is actually a bug. I followed the concept from the initialize stage ([here](https://github.com/NOAA-OWP/cfe/blob/master/src/bmi_cfe.c#L1477)), what looks to me is that they tried to start the index at 1 ([see](https://github.com/NOAA-OWP/cfe/blob/master/src/bmi_cfe.c#L1240)), but the BMI `Set_value` does not work this way, so we need to fix it.